qiskit/algorithms/evolvers/real/trotterization/trotter_qrte.py

```diff
@@ -162,6 +162,11 @@ def evolve(self, evolution_problem: EvolutionProblem) -> EvolutionResult:
                 "``quantum_instance`` was provided."
             )
         hamiltonian = evolution_problem.hamiltonian
+        if not isinstance(hamiltonian, (Pauli, PauliOp, SparsePauliOp, PauliSumOp, SummedOp)):
+            raise ValueError(
+                f"TrotterQRTE only accepts Pauli | PauliOp | SparsePauliOp | "
+                f"PauliSumOp | SummedOp, {type(hamiltonian)} provided."
+            )
         if not isinstance(hamiltonian, SparsePauliOp):  # TODO can we handle it better?
             hamiltonian = hamiltonian.bind_parameters(evolution_problem.hamiltonian_value_dict)
         if isinstance(hamiltonian, SummedOp):
```

Comment:
>         if not isinstance(hamiltonian, SparsePauliOp):  # TODO can we handle it better?

Should this be `if isinstance(hamiltonian, OperatorBase):` now since only opflow operators have this support for parameters/binding as far as I am aware